internal/controller/pkg/revision/reconciler.go

```diff
@@ -570,7 +572,24 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	// Establish control or ownership of objects.
-	refs, err := r.objects.Establish(ctx, pkg.GetObjects(), pr, pr.GetDesiredState() == v1.PackageRevisionActive)
+	objects := pkg.GetObjects()
+	commonLabels := pr.GetCommonLabels()
+	for _, obj := range objects {
+		// convert to resource.Object to be able to access metadata
+		d, ok := obj.(resource.Object)
+		if !ok {
+			return reconcile.Result{}, errors.New(errConfResourceObject)
+		}
+		labels := d.GetLabels()
+		if labels != nil {
+			for key, value := range commonLabels {
+				labels[key] = value
+			}
+		} else {
+			d.SetLabels(commonLabels)
+		}
+	}
```

Comment:
Any reason not to do this as part of the establisher?

Reply:
I moved the labeling not a method of the establisher